reconcile/templating/rendering.py

```diff
@@ -0,0 +1,106 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from jinja2.sandbox import SandboxedEnvironment
+from pydantic import BaseModel
+from ruamel import yaml
+
+from reconcile.gql_definitions.templating.templates import TemplateV1
+from reconcile.utils.jsonpath import parse_jsonpath
+
+
+class Renderer(ABC):
+    template: TemplateV1
+    data: "TemplateData"
+
+    def __init__(self, template: TemplateV1, data: "TemplateData"):
+        self.template = template
+        self.data = data
+        self.jinja_env = SandboxedEnvironment()
+
+    def _get_variables(self) -> dict[str, Any]:
+        return self.data.variables
+
+    def _get_current(self) -> dict[str, Any]:
+        return self.data.current or {}
+
+    def _jinja2_render_kwargs(self) -> dict[str, Any]:
+        return {**self._get_variables(), "current": self._get_current()}
+
+    def _render_template(self, template: str) -> str:
+        return self.jinja_env.from_string(template).render(
+            **self._jinja2_render_kwargs()
+        )
+
+    @abstractmethod
+    def get_output(self) -> str:
+        pass
+
+    def get_target_path(self) -> str:
+        return self._render_template(self.template.target_path)
+
+    def should_render(self) -> bool:
+        return self._render_template(self.template.condition or "True") == "True"
+
+
+class FullRenderer(Renderer):
+    def get_output(self) -> str:
+        return self._render_template(self.template.template)
+
+
+class PatchRenderer(Renderer):
+    def get_output(self) -> str:
```

Comment:
What does this method? There are so many operations, but at the end you don't return any of the variables? Please also add a `__doc__` string.